### PR TITLE
scaffolder: update to `react-jsonschema-form` `v5`

### PR DESCRIPTION
**Pull request by benjdlambert**
Replaces #9852

### CODE REVIEW

**Comment by benjdlambert** [comment]
Need to work through the typescript issues here too, but just opened this so that theres progress on it.

**Comment by benjdlambert** [comment]
As discussed before - this is gonna ship rjsfv5 + rjsfv3 together for now, and the breaking changes will be encapsulated behind the `/next` scaffolder for now. Maybe we can also come up with a way to introduce these changes in a less breaking way too. But want to get this in so that we can have them in parallel :)

File: plugins/scaffolder/src/next/TemplateWizardPage/Stepper/Stepper.tsx [hunk 1]
```diff
@@ -93,7 +97,7 @@ export const Stepper = (props: StepperProps) => {
     const returnedValidation = await validation(formData);
 
     const hasErrors = Object.values(returnedValidation).some(i => {
-      return i.__errors.length > 0;
+      return i.__errors?.length! > 0 ?? false;
```
Comment (Rugvip) [inline plugins/scaffolder/src/next/TemplateWizardPage/Stepper/Stepper.tsx:100]:
```suggestion
     i.__errors?.length
```

Missing the explicit `> 0` check is a bit meh, but feeling it's worth it here

**Review by Rugvip — CHANGES_REQUESTED** [review]
🎉 , some small things

**Review by Rugvip — APPROVED** [review]

**Pull request merged**